bin/fallback-module-directory-webpack-plugin.js

```diff
@@ -0,0 +1,140 @@
+/*eslint-env node*/
+
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+
+// Note, this has some inspiration from the AliasPlugin and its implementation
+// @see https://github.com/webpack/enhanced-resolve/blob/v4.1.0/lib/AliasPlugin.js
+
+function startsWith( string, searchString ) {
+	const stringLength = string.length;
+	const searchLength = searchString.length;
+
+	// early out if the search length is greater than the search string
+	if ( searchLength > stringLength ) {
+		return false;
+	}
+	let index = -1;
+	while ( ++index < searchLength ) {
+		if ( string.charCodeAt( index ) !== searchString.charCodeAt( index ) ) {
+			return false;
+		}
+	}
+	return true;
+}
+
+module.exports = class FallbackModuleDirectoryWebpackPlugin {
+	constructor( search, replacement, alias ) {
+		this.search = search;
+		this.replacement = replacement;
+		this.alias = this.parseAlias( alias );
+		this.hooks = [
+			[ 'described-resolve', 'resolve' ],
+			[ 'file', 'resolve' ],
+		];
+		this.applyFallback = this.applyFallback.bind( this );
+		this.doApply = this.doApply.bind( this );
+	}
+
+	parseAlias( alias ) {
+		if ( typeof alias === 'object' && ! Array.isArray( alias ) ) {
+			alias = Object.keys( alias ).map( ( key ) => {
+				let onlyModule = false;
+				let obj = alias[ key ];
+				if ( /\$$/.test( key ) ) {
+					onlyModule = true;
+					key = key.substr( 0, key.length - 1 );
+				}
+				if ( typeof obj === 'string' ) {
+					obj = {
+						alias: obj,
+					};
+				}
+				obj = Object.assign(
+					{
+						name: key,
+						onlyModule,
+					},
+					obj
+				);
+				return obj;
+			} );
+		}
+		return alias;
+	}
+
+	applyFallback( path ) {
+		// if directory does not have search string then bail
+		if ( ! path.includes( this.search ) ) {
+			return path;
+		}
+
+		if ( ! fs.existsSync( path ) ) {
+			path = path.replace( this.search, this.replacement );
```

Comment:
Actually, no this conditional style was intentional because the very first condition bails early if the search string isn't even present.  In other words if `legacy` isn't in the path, then there's no need to do a replacement (or to check if the path exists).  It's a minor optimization thing but I don't think I need to change this.

Reply:
doh, no you're right, I read the conditional wrong.  See change in 322f074